core/src/main/java/com/linecorp/armeria/common/metric/EventLoopMetrics.java

```diff
@@ -0,0 +1,134 @@
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import com.linecorp.armeria.common.util.Ticker;
+import com.linecorp.armeria.internal.common.metric.MicrometerUtil;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.SingleThreadEventLoop;
+import io.netty.util.concurrent.EventExecutor;
+
+/**
+ * A {@link MeterBinder} to observe Netty {@link EventLoopGroup}s. The following stats are currently
+ * exported per registered {@link MeterIdPrefix}.
+ *
+ * <ul>
+ *   <li>"event.loop.num.workers" (gauge) - the total number of Netty's event loops</li>
+ *   <li>"event.loop.pending.tasks" (gauge)
+ *     - the total number of IO tasks waiting to be run on event loops</li>
+ * </ul>
+ **/
+final class EventLoopMetrics implements MeterBinder {
+
+    private static final long UPDATE_INTERVAL_NANOS = TimeUnit.SECONDS.toNanos(3);
+
+    private final MeterIdPrefix idPrefix;
+    private final EventLoopGroup eventLoopGroup;
+
+    /**
+     * Creates an instance of {@link EventLoopMetrics}.
+     */
+    EventLoopMetrics(String name, EventLoopGroup eventLoopGroup) {
+        idPrefix = new MeterIdPrefix("armeria.netty").append(requireNonNull(name, "name"));
+        this.eventLoopGroup = requireNonNull(eventLoopGroup, "eventLoopGroup");
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        final Self metrics = MicrometerUtil.register(registry, idPrefix, Self.class, Self::new);
+        metrics.add(eventLoopGroup);
+    }
+
+    /**
+     * An actual implementation of {@link EventLoopMetrics}.
+     */
+    static final class Self {
+        private final Set<EventLoopGroup> registry = ConcurrentHashMap.newKeySet(2);
+        private final Ticker ticker;
+
+        private volatile double lastPendingTasks;
+        private volatile double lastNumWorkers;
+        private volatile long lastStatsUpdateTime;
+
+        Self(MeterRegistry parent, MeterIdPrefix idPrefix) {
+            this(parent, idPrefix, Ticker.systemTicker());
+        }
+
+        Self(MeterRegistry parent, MeterIdPrefix idPrefix, Ticker ticker) {
+            this.ticker = requireNonNull(ticker, "ticker");
+
+            final String numWorkers = idPrefix.name("event.loop.workers");
+            parent.gauge(numWorkers, idPrefix.tags(), this, Self::numWorkers);
+
+            final String pendingTasks = idPrefix.name("event.loop.pending.tasks");
+            parent.gauge(pendingTasks, idPrefix.tags(), this, Self::pendingTasks);
+        }
+
+        // This runs every 3 seconds at most.
```

Comment:
I think I carried over this pattern from the Caffeine metrics, which are admittedly more expensive to collect. I'm happy to drop the caching if that also works for @ikhoon and @minwoox. I don't want to do before we have a consensus.

Reply:
That's a good point. 👍 
I thought caching is okay because this type of information does not need to be immediately reflected in actual values.
Of course, we can revisit this later if we need to fix it. 😄

Reply:
I'd down to simplify it. I think pendingTask is pretty cheap to run. Let me make the change.

Reply:
Updated! PTAL

Reply:
Still LGTM. Thanks!